handout/di/angular2/avoiding_collisions_opaque_token.md

```diff
@@ -0,0 +1,98 @@
+# Avoiding Injection Collisions: OpaqueToken
+
+Since Angular allows the use of tokens as identifiers to its dependency injection system, one of the potential issues is using the same token to represent different entities. If, for example, the string `'token'` is used to inject an entity, it's possible that something totally unrelated also uses `'token'` to inject a different entity. When it comes time for Angular to resolve one of these entities, it might be resolving the wrong one. This behaviour might happen rarely, or are easy to resolve when it happens within a small team, but when it comes to multiple teams working separately on the same codebase or integration of 3rd party modules, the cost of these collisions goes up.
```

Comment:
> This behaviour might happen rarely, or are easy to resolve when it happens
should read or is easy, also as much as I love Canadian spelling behaviour needs to be behavior because of the style guide.